### PR TITLE
Cleanup: Removed unnecessary folders in react-datetime and highlight.…

### PR DESCRIPTION
**Pull request by HCanArslan**
…js dependency

### Description
This pull request addresses the task of removing unnecessary folders and dependencies from the react-datepicker package. The following changes have been made:

Deleted the following folders: docs/, docs-site/, examples/
Added them to the .gitignore file to ensure they are ignored by version control.
Removed highlight.js from the package's dependencies.
 
### Issues Resolved
This PR resolves the issue related to removing unnecessary files and dependencies from the react-datepicker package.
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] All tests pass
  - [x] `yarn lint`
  - [x] `yarn test-unit`
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/oui/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,14 +12,16 @@ docs/
 lib/
 es/
 test-env/
-.idea
```
Comment (BSFishy) [inline .gitignore:16]: Any reason this was removed?

Comment (HCanArslan) [reply .gitignore:16]: I think it was a mistake, sorry about that fixing it now

Comment (HCanArslan) [reply .gitignore:16]: no it was not a mistake because ".idea" exists on the top, am i correct?

Comment (BSFishy) [reply .gitignore:16]: You're right, I didn't see that :)

File: .gitignore [hunk 2]
```diff
@@ -13,14 +12,16 @@ docs/
...
 .vscode/
 .DS_Store
 .eslintcache
 .yo-rc.json
 npm-debug.log
 yarn-error.log
 .cache-loader
+/docs
+/docs-site
+/examples
```
Comment (BSFishy) [inline .gitignore:24]:
```suggestion
```

Upon further inspection, it looks like these ignores aren't actually needed.

**Comment by SergeyMyssak** [comment]
Hello @HCanArslan, could you please add a link to the issue in the description of the PR that this PR solves - https://github.com/opensearch-project/oui/issues/766?

**Comment by joshuarrrr** [comment]
Setting to draft for now. @HCanArslan feel free to move out of draft status when you're available to work on this again.

**Comment by joshuarrrr** [comment]
Superseded by https://github.com/opensearch-project/oui/pull/1067

**Pull request closed without merging**